corehq/apps/hqadmin/management/commands/prune_elastic_indices.py

```diff
@@ -0,0 +1,54 @@
+from optparse import make_option
+from django.conf import settings
+from django.core.management import BaseCommand
+from elasticsearch import Elasticsearch
+from pillowtop import get_all_pillows
+from pillowtop.listener import AliasedElasticPillow
+
+
+class Command(BaseCommand):
+    help = 'Delete all unreferenced elasticsearch indices.'
+
+    option_list = BaseCommand.option_list + (
+        make_option('--verbose', help='Additional logging.', action='store_true',
+                    default=False),
+        make_option('--noinput', help='Do not prompt user for input', action='store_true',
+                    default=False),
+    )
+
+    def handle(self, *args, **options):
+        es = Elasticsearch([{
+            'host': settings.ELASTICSEARCH_HOST,
+            'port': settings.ELASTICSEARCH_PORT,
+        }])
+        # call this before getting existing indices because apparently getting the pillow will create the index
+        # if it doesn't exist
+        found_indices = set(es.indices.get_aliases().keys())
+        existing_indices = set(
+            pillow.es_index for pillow in filter(lambda x: isinstance(x, AliasedElasticPillow), get_all_pillows())
+        )
+
+        if options['verbose']:
+            if existing_indices - found_indices:
+                print 'the following indices were not found:\n{}\n'.format(
+                    '\n'.join(existing_indices - found_indices)
+                )
```

Comment:
I suppose this could happen if you had just changed the mapping file, but hadn't yet started the reindex.  That seems pretty unlikely, but still, it might be a good idea to fail hard here, lest you delete the latest working index due to having a different/bad code version.